components/shared/time-diff.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { getTimeDifferenceInMinutes } from '@/utils/utils';
+import { TimeIcon } from 'clada-storybook';
+import React, { useState, useEffect } from 'react';
+
+export const TimeDiff = ({
+  postId,
+  text,
+}: {
+  postId: string;
+  text?: string;
+}) => {
+  const [timeDiff, setTimeDiff] = useState('');
```

Comment:
Comment only for the future: Probably something that can be easily done on the server side.